cpp/libs/bisect_nmoscpp/lib/src/nmos_controller.cpp

```diff
@@ -548,60 +569,52 @@ maybe_ok nmos_controller_t::modify_connection_receiver(const nmos_receiver_t& co
 
 expected<nmos::resource> nmos_controller_t::find_resource(const nmos::id& id)
 {
-    const auto resource_it = nmos::find_resource(base_controller_.node_model_.node_resources, id);
-    BST_ENFORCE(resource_it != base_controller_.node_model_.node_resources.end(),
-                "trying to find a non-existing NMOS resource {}", utility::us2s(id));
-    return *resource_it;
+    return find_resource_after_(delay_millis, base_controller_.node_model_.node_resources, id);
 }
 
 expected<nmos::resource> nmos_controller_t::find_connection_resource(const nmos::id& id)
 {
-    const auto resource_it = nmos::find_resource(base_controller_.node_model_.connection_resources, id);
-    BST_ENFORCE(resource_it != base_controller_.node_model_.connection_resources.end(),
-                "trying to find a non-existing NMOS connection resource {}", utility::us2s(id));
-    return *resource_it;
+    return find_resource_after_(delay_millis, base_controller_.node_model_.connection_resources, id);
 }
 
 maybe_ok nmos_controller_t::erase_resource(const nmos::id& resource_id)
 {
 
     BST_CHECK(erase_resource_after_(delay_millis, base_controller_.node_model_.node_resources, resource_id));
+    auto resource = find_resource(resource_id);
 
-    const auto resource = nmos::find_resource(base_controller_.node_model_.node_resources, resource_id);
-    BST_ENFORCE(base_controller_.node_model_.node_resources.end() == resource,
-                "NMOS Resource with id {} was not deleted", utility::us2s(resource_id));
+    if(!is_error(resource))
```

Comment:
Isn't the condition reversed here? I think it should be 
```
if(is_error(resource))
```

Reply:
I believe it is not reserved, because we are erasing a resource, and if we can still find it afterward, it means something went wrong with the erasing.